### ISSUE TITLE
ListView.separated causes wrong spacing when the separatorBuilder returns a Divider

### ISSUE THREAD

**Issue by dev975** [flutter/flutter#25940]
When the ListView.separated is used with a separatorBuilder function that returns a divider widget, the margin of the first item is asymmetric. I think the divider widget has a margin. Since there's no divider over the first item, but there's one between the first and the second item, the effect is an asymmetric margin of the first item. The workaround I found is to use a container instead of a divider.

Example using divider
```dart
ListView.separated(
        itemBuilder: _buildItem,
        itemCount: items.length,
        separatorBuilder: (BuildContext context, int index) => Divider(color: Colors.black12,))
```

![withdivider](https://user-images.githubusercontent.com/10226301/50572256-0c20ce00-0dbc-11e9-9d17-052334c68134.jpg)

Example using container
```dart
ListView.separated(
        itemBuilder: _buildItem,
        itemCount: items.length,
        separatorBuilder: (BuildContext context, int index) => Container(
            height: 1, width: double.infinity, color: Color(0xFFEFEFEF)))
```

![withcontainer](https://user-images.githubusercontent.com/10226301/50572257-117e1880-0dbc-11e9-81e7-973c3d0b88d4.jpg)

Here's my flutter doctor result
```
>flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Version 10.0.17763.195], locale en-GB)
[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
[√] Android Studio (version 3.2)
[√] VS Code (version 1.30.1)
[√] Connected device (1 available)

• No issues found!
```

**Comment by MichaelPriebe** [comment]
Divider has some padding, you can use `Divider(height: 0.0)` to have the same effect as the container.

**Comment by zoechi** [comment]
@dev975 does this solve your issue?

Please check https://flutter.io/docs/testing/debugging#visual-debugging

Please consider asking support questions in one of the other channels listed at http://flutter.io/support .

I'm going to close. If you still think this is a Flutter bug, please add a comment and I'll reopen.

**Comment by dev975** [comment]
Yes, I fixed using `Divider(height: 0.0)`. Thank you very much!
